crates/walrus-service/src/backup/service.rs

```diff
@@ -364,63 +364,62 @@ fn start_db_metrics_loop(metrics_runtime: &MetricsAndLoggingRuntime, config: &Ba
     let backup_db_metric_set = BackupDbMetricSet::new(&metrics_runtime.registry);
     let database_url = config.db_config.database_url.clone();
 
+    let stats_query = format!(
+        "
+            SELECT
+                state AS name,
+                COUNT(*)::bigint AS value
+            FROM blob_state
+            WHERE
+                state = 'archived' AND
+                COALESCE(end_epoch > (SELECT MAX(epoch) FROM epoch_change_start_event),
+                            FALSE)
+            GROUP BY 1
+            UNION
+            SELECT
+                state AS name,
+                COUNT(*)::bigint AS value
+            FROM blob_state
+            WHERE
+                state = 'deleted' OR
+                (state = 'waiting' AND
+                    COALESCE(end_epoch > (SELECT MAX(epoch) FROM epoch_change_start_event), TRUE))
```

Comment:
Is this for blobs explicitly deleted? Do we have a grace period for those blobs explicitly deleted?

Reply:
We do not handle explicitly deleted blobs in the backup system. It only tracks MAX(end_epoch) across the various incoming objects that point to the same blob id.